### PR TITLE
REF: insert self.on column _after_ concat

### PR DESCRIPTION
**Pull request by jbrockmendel**
The idea here is to push towards #34714 by making _wrap_results do things in the same order as other other similar methods do.  cc @mroeschke LMK if there is a simpler way to accomplish this.

Orthogonal to #35470, #35730, #35696.

### CODE REVIEW

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -402,36 +402,27 @@ def _wrap_results(self, results, blocks, obj, exclude=None) -> FrameOrSeries:
...
+        # if we have an 'on' column we want to put it back into
+        # the results in the same location
+        if self.on is not None and not self._on.equals(obj.index):
+            name = self._on.name
+            extra_col = Series(self._on, index=obj.index, name=name)
+            if name not in df.columns and name not in df.index.names:
+                new_loc = len(df.columns)
+                df.insert(new_loc, name, extra_col)
+            elif name in df.columns:
+                # TODO: sure we want to overwrite results?
```
Comment (jbrockmendel) [inline pandas/core/window/rolling.py:423]: @mroeschke can you comment on why we do this?

**Review by jreback — APPROVED** [review]
lgtm. ex comment by @mroeschke

File: pandas/core/window/rolling.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/window/rolling.py:423]: I haven't really touched `_wrap_results` much so can't really speak to why we do this. What's an example where this is overwriting results?

Comment (jbrockmendel) [reply pandas/core/window/rolling.py:423]: i dont have an example on hand, but can find one bc i know this line is covered by tests.  ATM im inclined to not-worry about it

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**